reconcile/terraform_users.py

```diff
@@ -210,7 +212,30 @@ def run(
 
 
 def early_exit_desired_state(*args, **kwargs) -> dict[str, Any]:
+    def add_account_identity(acc):
+        acc[IDENTIFIER_FIELD_NAME] = acc["path"]
+        return acc
+
+    def add_role_identity(role):
+        role[IDENTIFIER_FIELD_NAME] = role["name"]
+        return role
+
     return {
-        "accounts": queries.get_aws_accounts(terraform_state=True),
-        "roles": get_tf_roles(),
+        "accounts": [
+            add_account_identity(a)
+            for a in queries.get_aws_accounts(terraform_state=True)
+        ],
+        "roles": [add_role_identity(r) for r in get_tf_roles()],
     }
+
+
+def desired_state_shard_config() -> DesiredStateShardConfig:
+    return DesiredStateShardConfig(
+        shard_arg_name="account_name",
+        shard_path_selectors={
+            "accounts[*].name",
```

Comment:
Just for my understanding:

this is required to cover `resetPasswords` property right?

Reply:
yes, but also if the terraform state configuration (what bucket to use etc) changes for an account.
the assumption is that every queries field is queried for a purpose and is therefore part of the desired state.